code/game/machinery/washing_machine.dm

```diff
@@ -156,24 +237,24 @@ GLOBAL_LIST_INIT(dye_registry, list(
 		M.Translate(rand(-3, 3), rand(-1, 3))
 		animate(src, transform=M, time=2)
 
-/obj/machinery/washing_machine/wash(clean_types)
-	. = ..()
-	if(!busy && bloody_mess && (clean_types & CLEAN_TYPE_BLOOD))
-		bloody_mess = FALSE
-		update_icon()
-		. = TRUE
-
 /obj/machinery/washing_machine/proc/wash_cycle()
 	for(var/X in contents)
 		var/atom/movable/AM = X
 		AM.wash(CLEAN_WASH)
 		AM.machine_wash(src)
-
-	busy = FALSE
+		if(!color_source && isitem(X))
+			var/obj/item/I = X
+			I.appearance_change(I)
+			I.desc = initial(I.desc)
```

Comment:
Might be better to do this in `appearance_change`

Reply:
Done!